### ISSUE TITLE
Database is locked by another process

### ISSUE THREAD

**Issue by mac230** [djcb/mu#1518]
## Expected or desired behavior

I would like to use mu4e to view/manage/read email. 

## Actual behavior

Calling 'mu4e' always yields the following error message (even after system and emacs restart):

```
error in process sentinel: mu4e~proc-sentinel: Database is locked by another process
error in process sentinel: Database is locked by another process
```

## Steps to reproduce

Call 'mu4e' in emacs. 

## Versions of mu, mu4e/emacs, operating system etc.
mu/mu4e 1.2
GNU Emacs 25.3.50.2 (x86_64-pc-linux-gnu, GTK+ Version 3.22.30) 
ubuntu 18.04

## Any other detail
I followed the instructions at:
https://groups.google.com/forum/#!topic/mu-discuss/EncdNPKhUYo
https://github.com/djcb/mu/issues/8

The solution:

```
rm -rf ~/.mu/xapian && mu index
```

didn't change anything for me.  

I can run the following from the command line:

```
mu index -m 
mu index --rebuild -m 
```

w/o an error message.  

I can also run 'mu4e-update-mail-and-index' w/o an error message.  

So, basically, I can't get to the mu4e headers.  I've been using mu4e on a mac for several years and am using essentially the same config on this system.  

There were a couple of discussion of possible .lock files.  Could you elaborate on what to look for.  On my system:

```
find /home/ -iname "*\.lock*"
```

shows the following:

```
/home/mahlon/.emacs.d/.emacs.desktop.lock
/home/mahlon/anaconda3/pkgs/jupyterlab-1.1.4-pyhf63ae98_0/site-packages/jupyterlab/staging/yarn.lock
/home/mahlon/anaconda3/lib/python3.7/site-packages/jupyterlab/staging/yarn.lock
```

I don't think any of these are related to mu/mu4e.

Any help much appreciated.  Thanks.

**Comment by djcb** [comment]
Hmm, _something_ must have opened the database, that's why you're getting that error message. What does 
```
lsof  2>/dev/null | grep xapian
```
give you when you run it _before_ you start mu4e?

**Comment by mac230** [comment]
No output of that command before or after starting mu4e.  

Forgot to mention that I'm using mbsync and msmtp w/ mu/mu4e as well.  

Tried removing all messages and re-building the mail dir w/ no results after reading the discussion here:
https://github.com/djcb/mu/issues/198

**Comment by djcb** [comment]
Rebuilding etc. won't really help here, I think. 

It's about another process having the xapian database open for writing which makes mu4e error out. For instance you cannot run `mu4e` while running `mu index` (or vice-versa).

So perhaps try immediately after you get that error:
`lsof +D /home/mahlon/.mu`

Perhaps you have some other script running `mu index` in the background?

**Comment by mac230** [comment]
No output from

```
lsof +D /home/mahlon/.mu
```

I tried:

```
lsof +D ~/.cache/mu
```

which returned the following:

```
COMMAND  PID   USER   FD   TYPE DEVICE SIZE/OFF     NODE NAME
bash    7335 mahlon  cwd    DIR  253,1     4096 10094321 .
lsof    8226 mahlon  cwd    DIR  253,1     4096 10094321 .
lsof    8227 mahlon  cwd    DIR  253,1     4096 10094321 .
```

Also, no other scripts running mu index.

**Comment by mac230** [comment]
I deleted everything I could find related to mu and mu4e and rebuilt/reinstalled.   It seems to be working now.  I couldn't track what was actually causing the issue, but am closing since I have the desired behavior now.  Thanks.

**Comment by djcb** [comment]
Hmm good that it works, even if we don't know why!

If it ever happens again, perhaps check if the .lock exists _before_ starting mu4e.

**Comment by johnhamelink** [comment]
> ```
> lsof +D ~/.cache/mu
> ```
> 
> which returned the following:
> 
> ```
> COMMAND  PID   USER   FD   TYPE DEVICE SIZE/OFF     NODE NAME
> bash    7335 mahlon  cwd    DIR  253,1     4096 10094321 .
> lsof    8226 mahlon  cwd    DIR  253,1     4096 10094321 .
> lsof    8227 mahlon  cwd    DIR  253,1     4096 10094321 .
> ```

Once I knew the process ID, all I had to do was kill it, run mbsync, and then run mu4e and all was well again - I thought I'd add a comment here for anyone who comes across this issue in future.

**Comment by titanofold** [comment]
> Once I knew the process ID, all I had to do was kill it, run mbsync, and then run `M-x mu4e` as normal, and all was well again - I thought I'd add a comment here for anyone who comes across this issue in future.

Same-ish!

In my case, there was a phantom Emacs session still running that needed to be killed. (WSL fun.) All I had to do was kill the old Emacs process and then fire up Emacs and mu4e again. No extra steps.
